src/utils/datetime/validator.test.ts

```diff
@@ -0,0 +1,147 @@
+import {isValid} from './validator'
+
+describe('the datetime validator', () => {
+  it('should return true on valid date formats', function() {
+    expect(isValid('1999-02-09 23:00:00', 'YYYY-MM-DD HH:mm:ss')).toBeTruthy()
+    expect(isValid('1999-02-09', 'YYYY-MM-DD')).toBeTruthy()
+    expect(isValid('1999-02-09 23:00', 'YYYY-MM-DD HH:mm')).toBeTruthy()
+    expect(isValid('1999-02-09 23:00:00', 'YYYY-MM-DD HH:mm:ss')).toBeTruthy()
+    expect(
+      isValid('1999-02-09 23:00:00.000', 'YYYY-MM-DD HH:mm:ss.sss')
+    ).toBeTruthy()
+    expect(
+      isValid('1999-02-09 12:00:00 PM -0700', 'YYYY-MM-DD hh:mm:ss a ZZ')
```

Comment:
Here, we are testing for local timezone, but it's okay because we provide the timezone. There will be no conflict in the CI.